docs/essentials/controls.md

```diff
@@ -45,9 +87,57 @@ By default, Storybook will choose a control for each arg based on its initial va
 
 </IfRenderer>
 
+<IfRenderer renderer='ember'>
+
+By default, Storybook will try to infer the required argTypes and associated controls for your stories based on the metadata provided by the [`@storybook/ember-cli-storybook`](https://github.com/storybookjs/ember-cli-storybook) addon. You'll need to take some additional steps to set it up properly.
+
+Update your `ember-cli-build.js` configuration file to include the addon.
+
+<!-- prettier-ignore-start -->
+
+<CodeSnippets
+  paths={[
+   'ember/storybook-ember-cli-build.js.mdx',
+  ]}
+/>
+
+<!-- prettier-ignore-end -->
+
+Restart your application to generate the metadata file (i.e., `storybook-docgen/index.json`) and update your `.storybook/preview.js` file to include it, which will be used to create the controls and argTypes for your stories.
+
+<!-- prettier-ignore-start -->
+
+<CodeSnippets
+  paths={[
+   'ember/storybook-preview-custom-metadata.js.mdx',
+  ]}
+/>
+
+<!-- prettier-ignore-end -->
+
+<Callout variant="info">
+
+Enabling this feature will generate a `storybook-docgen/index.json` automatically with each build. For more information on how the metadata is generated, refer to [documentation](https://github.com/storybookjs/storybook/tree/next/code/frameworks/ember) for the Ember framework.
+
+</Callout>
+
+When you set the `component` annotation to the default export of your story file, it will be used to infer the controls and auto-generate the matching [`argTypes`](../api/arg-types.md) for your component.
```

Comment:
```suggestion
When you set the `component` annotation of the default export of your story file, it will be used to infer the controls and auto-generate the matching [`argTypes`](../api/arg-types.md) for your component.
```